base/linalg/generic.jl

```diff
@@ -795,6 +795,19 @@ function inv(A::AbstractMatrix{T}) where T
     A_ldiv_B!(factorize(convert(AbstractMatrix{S}, A)), eye(S0, checksquare(A)))
 end
 
+function pinv(v::AbstractVector{T}, tol::Real=real(zero(T))) where T
```

Comment:
A vector has one singular value s, which obvious is the maximum one. That singular value will be ignored (resulting in a zero vector)  if s<=s*tol, i.e. tol>=1 or s=0.